### ISSUE TITLE
Capable Captain does not attach to some merchant ships

### ISSUE THREAD

**Issue by MrZorG33** [We-the-People-civ4col-mod/Mod#948]
It is impossible to attach a Capable Captain to the following ships: Brigantine, Whaling Ship, Merchant Ship, Flute, Caravel, Coastal Ships, Smuggler Ship, Slave Barque. all these ships belong to the UNITCOMBAT_NAVAL_TRADE class.
![image](https://github.com/We-the-People-civ4col-mod/Mod/assets/61798883/4f093474-8df7-4f8d-bb11-8ce8e8e8f9b4)

**Comment by johan-buret** [comment]
Please confirm that you can attach a capitain to the following ships : Galleon, Carrack, and Caravela Redonda.

If that's the case, it means that there is something with the Defense Only ships that prevent them to be led by a captain, which is wrong. All ships should be led by a captain.

**Comment by MrZorG33** [comment]
Yes, you can attach a captain to the ships you specify.

**Comment by MrZorG33** [comment]
comparing the Galleon and the Slave Barque (description in xml) I did not find any difference (except for the fact that one carries treasures and the other slaves)

**Comment by johan-buret** [comment]
All the units you have listed have this flag set:
`` <bOnlyDefensive>1</bOnlyDefensive>``

which shows in Colopedia as such :

![image](https://github.com/We-the-People-civ4col-mod/Mod/assets/7459821/3db68019-9a68-4adb-b338-492a42fdaead)

The naval trade ships that have some attack capabilities can get a captain on board. Those who are  defensive only cannot. This is the bug.

**Comment by raystuttgart** [comment]
There is logic that prevents Admirals / Captains / Generals / Lieutenants to accidentally be wasted to Units that cannot attack.
It was implemented **on purpose** because there was a user complaint in the past about it.

Same goes for XP of Units on the Plot.
It is only given to Units with certain pre-conditions.

The chage was originally implemented / suggested by **Ramstormp**.
We then picked it up, merged and further improved it.

**Summary:**
It seems it is all working as designed.
(No bug.)
